crates/uv-python/src/interpreter.rs

```diff
@@ -608,6 +608,29 @@ impl Interpreter {
     }
 }
 
+/// Calls `fs_err::canonicalize` on Unix. On Windows, avoids attempting to resolve symlinks
+/// but will resolve junctions if they are part of a trampoline target.
+pub fn canonicalize_executable(path: impl AsRef<Path>) -> std::io::Result<PathBuf> {
+    let path = path.as_ref();
+    debug_assert!(
+        path.is_absolute(),
+        "path must be absolute: {}",
+        path.display()
+    );
```

Comment:
Why must the path be absolute?

Reply:
This method is moved from `uv_fs` and this was an existing assumption.

Reply:
I don't understand why the existing assumption was there though, I think it's because otherwise it would be surprising behavior on Windows but you changed the Windows behavior here.

Reply:
This can still behave in the old way if the executable is not a trampoline. I can look into why this `debug_assert` was added in the first place, but I don't think my changes would have changed the invariant for the non-trampoline case.